angular/index.d.ts

```diff
@@ -97,7 +97,7 @@ declare namespace angular {
          * @param iterator Iterator function.
          * @param context Object to become context (this) for the iterator function.
          */
-        forEach<T>(obj: T[], iterator: (value: T, key: number) => any, context?: any): any;
+        forEach<T>(obj: T[], iterator: (value: T, key: number, obj?: T[]) => any, context?: any): any;
```

Comment:
It should be `obj`, not `obj?`. See http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#optional-parameters-in-callbacks
Also please change the return types for both the callback (should be `void`) and `forEach` (it returns `obj`).